tango/integrations/gs/workspace.py

```diff
@@ -170,6 +173,170 @@ def registered_runs(self) -> Dict[str, Run]:
 
         return runs
 
+    def search_registered_runs(
+        self,
+        *,
+        sort_by: Optional[RunSort] = None,
+        sort_descending: bool = True,
+        match: Optional[str] = None,
+        start: int = 0,
+        stop: Optional[int] = None,
+    ) -> List[str]:
+        run_entities = self._fetch_run_entities(
+            sort_by=sort_by, sort_descending=sort_descending, match=match, start=start, stop=stop
+        )
+        return [e.key.name for e in run_entities]
+
+    def num_registered_runs(self, *, match: Optional[str] = None) -> int:
+        count = 0
+        for _ in self._fetch_run_entities(match=match):
+            count += 1
+        return count
+
+    def _fetch_run_entities(
+        self,
+        *,
+        sort_by: Optional[RunSort] = None,
+        sort_descending: bool = True,
+        match: Optional[str] = None,
+        start: int = 0,
+        stop: Optional[int] = None,
+    ) -> Generator[datastore.Entity, None, None]:
+        from itertools import islice
+
+        # Note: we can't query or order by multiple fields without a suitable
+        # composite index. So in that case we have to apply remaining filters
+        # or slice and order locally. We'll default to using 'match' in the query.
+        # But if 'match' is null we can sort with the query.
+        sort_locally = match is not None
+
+        sort_field: Optional[str] = None
+        if sort_by == RunSort.START_DATE:
+            sort_field = "start_date"
+        elif sort_by == RunSort.NAME:
+            sort_field = "name"
+        elif sort_by is not None:
+            raise NotImplementedError(sort_by)
+
+        order: List[str] = []
+        if sort_field is not None and not sort_locally:
+            order = [sort_field if not sort_descending else f"-{sort_field}"]
+
+        query = self._ds.query(kind="run", order=order)
+        if match is not None:
+            # HACK: Datastore has no direct string matching functionality,
+            # but we can assume that run names are alphanumeric. So this comparison
+            # is equivalent to checking if 'name' starts with 'match'.
+            query.add_filter("name", ">=", match)
+            query.add_filter("name", "<=", match + "~")
```

Comment:
Do we have to escape `match` in some way? What if it already contains a `~`?

Reply:
I don't think we need to, assuming run names shouldn't have a `~` in them. But I suppose instead of using `~` to hack this, we could the `DEL` character (that's the last ASCII character). b883ad0

Reply:
Oh, then wouldn't it be cleaner to do `..., "<", match[:-1] + chr(ord(match[-1]) + 1)`? Then it's not dependent on ASCII in any way.

Reply:
Yes good point, although I think `match + chr(ord(match[-1]) + 1)` is fine instead of `match[:-1] + ...`.

Reply:
2e84840

Reply:
> `match + chr(ord(match[-1]) + 1)` is fine instead of `match[:-1] + ...`

Really? So if `match` is `"cromulent"`, it'll be `< "cromulentu"`? I don't think that's correct. It should be `< "cromulenu"`.

Reply:
Otherwise it will exclude the string `"cromulentz"`, which definitely as a prefix of `"cromulent"`.

Reply:
Ah, you're right: 2857666